u_parrots.go

```diff
@@ -619,38 +633,86 @@ func (uconn *UConn) generateRandomizedSpec(WithALPN bool) (ClientHelloSpec, erro
 		}
 		p.Extensions = append(p.Extensions, &ks, &pskExchangeModes, &supportedVersionsExt)
 	}
-	err = shuffleTLSExtensions(p.Extensions)
-	if err != nil {
-		return p, err
-	}
+	r.shuffleTLSExtensions(p.Extensions)
 	err = uconn.SetTLSVers(p.TLSVersMin, p.TLSVersMax)
 	if err != nil {
 		return p, err
 	}
 
-	return p, nil
+	return p, r.savedErr
+}
+
+// utlsRandomizer never returns errors, but stores first one in savedErr
```

Comment:
Why doesn't this implement `math/rand/Source` (Or `Source64`?).  The runtime library's `math/rand/Rand` provides perfectly good implementations of most of the helpers given a `Source`, except for the biased coin flip, that's trivial to implement given `Rand.Float32()`.

Reply:
I don't see why it would need to implement any interfaces, it's not even exposed to user.
Tbh, I am not so in love with math/rand after I've seen `NewSource()` round down int64 seed to int32 internally, meaning that I have to verify every function of theirs. Kinda seemed easier to just make a few short functions myself, but it is true that some of those implementations are perfectly good